### PR TITLE
[tween] Add missing types for custom progress or easing in `props`.

### PR DESCRIPTION
**Pull request by dumganhar**
Re: #17110

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/scene-graph/node.ts [hunk 1]
```diff
@@ -1900,6 +1911,12 @@ export class Node extends CCObject implements ISchedulable, CustomSerializable {
      * @deprecated since v3.5.0, this is an engine private interface that will be removed in the future.
      */
     public _onBatchCreated (dontSyncChildPrefab: boolean): void {
+        if (this._eventMask & ACTIVE_ON) {
```
Comment (minggo) [inline cocos/scene-graph/node.ts]: Why need to check it here?

File: cocos/scene-graph/node.ts [hunk 2]
```diff
@@ -1920,8 +1937,13 @@ export class Node extends CCObject implements ISchedulable, CustomSerializable {
      * @deprecated since v3.5.0, this is an engine private interface that will be removed in the future.
      */
     public _onPostActivated (active: boolean): void {
+        if (this._eventMask & ACTIVE_ON) {
```
Comment (minggo) [inline cocos/scene-graph/node.ts]: Is post active the same as node active?

File: cocos/scene-graph/node.ts (hunk 1 above)
Comment (dumganhar) [reply cocos/scene-graph/node.ts]: For performance consideration, no all node instance need to dispatch `ACTIVE_CHANGED` events, so checking the mask here for the nodes that are used as `tween` targets .

File: cocos/scene-graph/node.ts (hunk 2 above)
Comment (dumganhar) [reply cocos/scene-graph/node.ts]: Yes, u could refer the code in `node-activator.ts`.
WeChat0ad0a025b8fa1547a1cf0db50b81c382

Comment (dumganhar) [reply cocos/scene-graph/node.ts]: Also you could refer which in Creator 2.x ( https://github.com/cocos/cocos-engine/blob/cedd59c725a76736295773586973e3360d5eaa17/cocos2d/core/CCNode.js#L1747 )

2.x also did ActionManager resume/pause in _onPostActivated

**Review by minggo — APPROVED** [review]

File: cocos/scene-graph/node.ts (hunk 2 above)
Comment (dumganhar) [reply cocos/scene-graph/node.ts]: Ref: https://github.com/cocos/cocos-engine/pull/17141

File: cocos/scene-graph/node.ts (hunk 1 above)
Comment (dumganhar) [reply cocos/scene-graph/node.ts]: Ref: https://github.com/cocos/cocos-engine/pull/17141

**Review by minggo — APPROVED** [review]

**Pull request merged**